test/unit/utils.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+const { assert } = require("chai");
+const { bigintToLong, longToBigint } = require("../../lib/new-utils");
+const Long = require("long");
+
+const values = [
+    "0",
+    "1",
+    "-1",
+    "12",
+    "-12",
+    "20040400020",
+    "120000000000",
+    "9223372036854775807",
+    "9223372036854775806",
+    "-9223372036854775808",
+    "-9223372036854775805",
+    "-20040400020",
+];
+
+describe("Numeric types conversions", function () {
+    // We assume BigInt and Long are fully correct
+    describe("BigInt to Long", function () {
+        it("should covert correctly", function () {
+            values.forEach((value) => {
+                let bigInt = BigInt(value);
+                let long = bigintToLong(bigInt);
+                assert.deepEqual(value, long.toString());
+            });
+        });
+    });
+    describe("Long to BigInt", function () {
+        it("should covert correctly", function () {
+            values.forEach((value) => {
+                let long = Long.fromString(value);
+                let bigInt = longToBigint(long);
+                assert.deepEqual(value, bigInt.toString());
```

Comment:
Is `deepEqual` a content-wise comparison between strings (as opposed to pointer-wise) ?

Reply:
I missread the docuemntation, and what I wanted was to compare the objects with ===, instead of == in normal equal.
When it comes to deep equal the [docuentation](https://www.chaijs.com/api/assert/) says: ``Asserts that actual is deeply equal to expected.`` and to be fair I don't fully understand what it means...